app/models/concerns/solr_indexable.rb

```diff
@@ -257,4 +257,9 @@ def ancestor_structure(ancestor_title)
   def generate_hash
     Digest::MD5.hexdigest oid.to_s
   end
+
+  def generate_pid(project_identifier, oid)
+    parent_object = ParentObject.find_by(oid: oid)
```

Comment:
Is parent_object the same as "self" here?  If you aren't explicitly reloading it, you could just access self&.project_identifier instead of loading the parent_object with find_by.

Reply:
self did work.  Thanks for helping me avoid a db call!